spec/lib/rmt/cli/smt_importer_spec.rb

```diff
@@ -0,0 +1,285 @@
+    end
+  end
+
+  describe '#import_repositories' do
+    before do
+      allow(script).to receive(:read_csv).with('enabled_repos').and_return enabled_repos
+    end
+
+    let(:repo_1) { create :repository }
+    let(:repo_2) { create :repository }
+
+    context 'with repository' do
+      let(:enabled_repos) { [repo_1.scc_id, repo_2.scc_id] }
+
+      it 'enables mirroring for given repositories' do
+        expect { script.import_repositories }.to output(<<-OUTPUT.strip_heredoc).to_stdout
+          Enabled mirroring for repository #{repo_1.scc_id}
+          Enabled mirroring for repository #{repo_2.scc_id}
+        OUTPUT
+
+        expect(repo_1.reload.mirroring_enabled).to be(true)
+        expect(repo_2.reload.mirroring_enabled).to be(true)
+      end
+    end
+
+
+    context 'without repository' do
+      let(:enabled_repos) { [0] }
+
+      it 'warns the user if the repository was not found' do
+        expect { script.import_repositories }.to output(/Repository 0, perhaps you no longer/).to_stderr
+      end
+    end
+  end
+
+  describe '#import_custom_repositories' do
+    before do
+      allow(script).to receive(:read_csv).with('enabled_custom_repos').and_return enabled_custom_repos
+    end
+
+    context 'already created repository' do
+      let(:repo) { create :repository, :custom }
+      let(:product_1) { create :product }
+      let(:product_2) { create :product }
+
+      let(:enabled_custom_repos) do
+        [
+          [product_1.id, repo.name, repo.external_url],
+          [product_2.id, repo.name, repo.external_url]
+        ]
+      end
+
+      it 'adds the association between repository and product if neccesary' do
+        expect { script.import_custom_repositories }.to output(<<-OUTPUT.strip_heredoc).to_stdout
+          Added association between #{repo.name} and product #{product_1.id}
+          Added association between #{repo.name} and product #{product_2.id}
+        OUTPUT
+        expect(RepositoriesServicesAssociation.find_by(service: product_1.service, repository: repo)).not_to be_nil
+        expect(RepositoriesServicesAssociation.find_by(service: product_2.service, repository: repo)).not_to be_nil
+      end
+    end
+
+    context 'repository not created yet' do
+      let(:product) { create :product }
+      let(:repo_name) { 'SAMPLE_REPO' }
+      let(:repo_url) { 'https://SAMPLE_REPO_URL/repo/...' }
+      let(:local_path) { '/srv/repos/SAMPLE_REPO_URL' }
+
+      let(:enabled_custom_repos) { [[product.id, repo_name, repo_url]] }
+
+      it 'creates the repository if not created' do
+        expect(Repository).to receive(:make_local_path).with(repo_url).and_return local_path
+
+        expect { script.import_custom_repositories }.to output(<<-OUTPUT.strip_heredoc).to_stdout
+          Added association between #{repo_name} and product #{product.id}
+        OUTPUT
+
+        expect(Repository.find_by(external_url: repo_url, local_path: local_path)).not_to be_nil
```

Comment:
Would be good to have `RepositoriesServicesAssociation` expectation (same as in previous example) as well.